easypost/shipment.py

```diff
@@ -161,3 +161,13 @@ def get_lowest_smartrate(smartrates, delivery_days: int, delivery_accuracy: str)
             raise Error(message="No rates found.")
 
         return lowest_smartrate
+
+    def retrieve_estimated_delivery_date(self, planned_ship_date: str) -> List[Dict[str, Any]]:
```

Comment:
Function returns rate objects, correct? Function name misleading.

Reply:
Kind of, it returns the original rate objects (nested) with a new `easypost_time_in_transit_data` object attached to each so it's not an actual list of Rate objects because it's a list of objects that contain both a `rate` and an `easypost_time_in_transit_data` object inside them.

As for the function name, it matches the API docs provided which is the get the estimated delivery date for rates. Does that work? https://www.easypost.com/docs/api#retrieve-estimated-delivery-date-and-total-transit-days-across-all-rates-for-a-shipment

Reply:
Sounds good